packages/affiliates/libs/models.js

```diff
@@ -171,13 +171,14 @@ function History() {
   const history = [];
   // Used internally, but will insert a block into cache sorted by timestamp ascending
   function insert(data) {
-    assert(data.blockNumber, "requires blockNumber");
+    assert(data.blockNumber === 0 || data.blockNumber > 0, "requires blockNumber");
     const index = lodash.sortedIndexBy(history, data, "blockNumber");
     history.splice(index, 0, data);
     return data;
   }
   function lookup(blockNumber) {
     const index = lodash.sortedIndexBy(history, { blockNumber }, "blockNumber");
+    assert(history.length, "history is empty");
```

Comment:
What if `history === undefined`, will this unexpectedly throw an error?

Reply:
history is defined as a const and given an empty array as the default value in construction so it should be ok